crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -907,6 +929,86 @@ impl CheckpointStore {
             .all(|opt| opt.is_some()))
     }
 
+    async fn wait_for_all_checkpoint_transactions_executed<A>(
```

Comment:
I think we are stepping on each other's toes a bit... see my draft PR: https://github.com/MystenLabs/sui/pull/2943

Let's chat about what you need this function to do. I'm not quite clear because it looks like it doesn't actually execute any certs.